### PR TITLE
Core: CSF v3 preview

### PR DESCRIPTION
**Pull request by shilman**
Issue: N/A

## What I did

- [x] Story object support
- [x] Feature flag for CSF v3
- [x] Global render function for React
- [x] Setup function support

## How to test

- [ ] See attached demo/tests

### CODE REVIEW

**Comment by tmeasday** [comment]
Maybe instead off the `implicitStoryFn` jazz we should add the concept of a global render function and have the framework set it in their preset?

We don't even necessarily need to let users define it globally just yet.

File: lib/core-client/src/preview/normalizeStory.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+const deprecatedStoryAnnotationWarning = deprecate(() => {}, deprecatedStoryAnnotation);
+
+/**
+ * Utilities for normalizing a story to support different
+ */
+export const normalizeV2 = (key: string, storyExport: any, meta: any, globalRender: any) => {
+  const exportType = typeof storyExport;
+  if (exportType !== 'function') {
+    logger.info(`Unexpected story export "${key}": expected function, received "${exportType}."`);
```
Comment (yannbf) [inline lib/core-client/src/preview/normalizeStory.ts:32]:
```suggestion
    logger.info(`Unexpected story export "${key}": expected function, received "${exportType}".`);
```

File: lib/core-client/src/preview/normalizeStory.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
+import { logger } from '@storybook/client-logger';
+import { ClientApi } from '@storybook/client-api';
+import { storyNameFromExport, toId } from '@storybook/csf';
+import dedent from 'ts-dedent';
+import deprecate from 'util-deprecate';
+import global from 'global';
+
+const { PREVIEW_CSF_V3 } = global;
+const isCsfV3 = () => {
+  try {
```
Comment (yannbf) [inline lib/core-client/src/preview/normalizeStory.ts:10]: why try catch?

File: lib/core-client/src/preview/normalizeStory.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  };
+
+  return {
+    name: storyName || exportName,
+    storyFn,
+    parameters,
+  };
+};
+
+export const normalizeV3 = (key: string, storyExport: any, meta: any, globalRender: any) => {
```
Comment (yannbf) [inline lib/core-client/src/preview/normalizeStory.ts:69]: Is it possible to type them better? Something like:
```suggestion
export const normalizeV3 = (key: string, storyExport: any, meta: Meta, globalRender: Story) => {
```

File: lib/core-common/src/types.ts [hunk 4]
```diff
@@ -235,12 +236,21 @@ export interface StorybookConfig {
...
     postcss?: boolean;
+
+    /**
+     * Build stories.json automatically on start/build
+     */
     buildStoriesJson?: boolean;
+
+    /**
+     * Activate preview of CSF v3.0
+     */
```
Comment (yannbf) [inline lib/core-common/src/types.ts:252]: Does it make sense to add a tsdoc [experimental](https://tsdoc.org/pages/tags/experimental/) flag?
On the same note, would be nice if `Meta` had its type updated to have the `render` property, also under experimental flag

**Review by tmeasday — APPROVED** [review]

**Pull request merged**